### PR TITLE
St386 - add properties, fields and notes from meta files

### PR DESCRIPTION
**Pull request by gtryus**
Allow Topic and Keyword as values without custom_ prefix
Use How to contact as actor address
Add actor, content, and resource descriptions
User UI language as analysis language
Put Code in actor's data

Card: https://trello.com/c/hgZrUmAB
Depends on: sillsdev/libpalaso#657

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/saymore/25)

### CODE REVIEW

File: src/SayMore/Model/ArchivingHelper.cs [hunk 1]
```diff
@@ -185,9 +186,11 @@ private static void AddIMDISession(Session saymoreSession, ArchivingDlgViewModel
 			imdiSession.SocialContext = GetFieldValue(sessionFile, "additional_Social_Context");
 			imdiSession.Task = GetFieldValue(sessionFile, "additional_Task");
 
+			imdiSession.AddContentDescription(new LanguageString{Value = GetFieldValue(sessionFile, "notes"), Iso3LanguageId = analysisLanguage});
+
 			// custom session fields
 			foreach (var item in saymoreSession.MetaDataFile.GetCustomFields())
-				imdiSession.AddContentKeyValuePair(item.FieldId, item.ValueAsString);
+				imdiSession.AddContentKeyValuePair(item.FieldId.Substring(XmlFileSerializer.kCustomFieldIdPrefix.Length), item.ValueAsString);
```
Comment (jasonleenaylor) [inline src/SayMore/Model/ArchivingHelper.cs]: The commit comment mentions some specific fields having the 'custom_' removed, but this code change removes it for all custom fields in the session and the person field. Is that desired?

**Comment by gtryus** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/SayMore/Model/ArchivingHelper.cs, line 193 at r1](https://reviewable.io/reviews/sillsdev/saymore/25#-L7zCepDqbbSO0yPFbSm-r1-193:-LBgjWF1-jCXy7Fre0QI:b6fv08p) ([raw file](https://github.com/sillsdev/saymore/blob/e4de9171829573a51ed8246e630c41fd374cf5f6/src/SayMore/Model/ArchivingHelper.cs#L193)):*
<details><summary><i>Previously, jasonleenaylor (Jason Naylor) wrote…</i></summary><blockquote>

The commit comment mentions some specific fields having the 'custom_' removed, but this code change removes it for all custom fields in the session and the person field. Is that desired?
</blockquote></details>

The IMDI export has a separate element Keys for all key fields. The prefix custom_ is a saymore internal construct which is not required or expected by the IMDI export schema.

---


*Comments from [Reviewable](https://reviewable.io/reviews/sillsdev/saymore/25)*

File: src/SayMoreTests/model/SessionArchivingTests.cs [hunk 2]
```diff
@@ -320,51 +322,223 @@ public void GetOneLanguage_UndeterminedIso_ReturnsNull()
...
+		}
+
+		public void AddIMDISession_AddSituation_AddSituationAsContentKey()
+		{
+			const string sampleSituation = "village hut recording";
+			_session._metaFile.Setup(f => f.GetStringValue("situation", It.IsAny<string>())).Returns(sampleSituation);
+			var model = AddIMDISessionTestSetup(out var imdiSession);
+			ArchivingHelper.AddIMDISession(_session, model.Object);
+			var situationValue = (from k in imdiSession.Object.MDGroup.Content.Keys.Key
+								  where k.Name == "Situation"       // They keyword in the saymore session is lower case but in Imdi we chose upper case
```
Comment (jasonleenaylor) [inline src/SayMoreTests/model/SessionArchivingTests.cs]: sp in comment

File: src/SayMoreTests/model/SessionArchivingTests.cs [hunk 3]
```diff
@@ -320,51 +322,223 @@ public void GetOneLanguage_UndeterminedIso_ReturnsNull()
...
+			sourceMediaFile1?.VerifyAll();
 		}
 		#endregion
 	}
 
 	public class DummySession : Session
 	{
 		public string[] _participants;
-		private readonly Mock<ProjectElementComponentFile> _metaFile = new Mock<ProjectElementComponentFile>();
+		public readonly Mock<ProjectElementComponentFile> _metaFile = new Mock<ProjectElementComponentFile>();
```
Comment (jasonleenaylor) [inline src/SayMoreTests/model/SessionArchivingTests.cs]: _ in front of a variable indicates that it is private. So this should be metaFile or MetaFile

**Comment by gtryus** [comment]
Review status: 0 of 2 files reviewed at latest revision, 4 unresolved discussions.

---

*[src/SayMore/Model/ArchivingHelper.cs, line 313 at r2](https://reviewable.io/reviews/sillsdev/saymore/25#-L7zCepDqbbSO0yPFbSm-r2-313:-LBgzM_61CmC00N2WcBD:b-896fix) ([raw file](https://github.com/sillsdev/saymore/blob/09455e50f85a104e4f38e297478798865189adb4/src/SayMore/Model/ArchivingHelper.cs#L313)):*
<details><summary><i>Previously, jasonleenaylor (Jason Naylor) wrote…</i></summary><blockquote>

tab
</blockquote></details>

Done.

---

*[src/SayMoreTests/model/SessionArchivingTests.cs, line 366 at r2](https://reviewable.io/reviews/sillsdev/saymore/25#-LBgj7QQB6owHaZa7jw0-r2-366:-LBgzOxxBWociGQqjmqU:b-896fix) ([raw file](https://github.com/sillsdev/saymore/blob/09455e50f85a104e4f38e297478798865189adb4/src/SayMoreTests/model/SessionArchivingTests.cs#L366)):*
<details><summary><i>Previously, jasonleenaylor (Jason Naylor) wrote…</i></summary><blockquote>

sp in comment
</blockquote></details>

Done.

---

*[src/SayMoreTests/model/SessionArchivingTests.cs, line 541 at r2](https://reviewable.io/reviews/sillsdev/saymore/25#-LBgj7QQB6owHaZa7jw0-r2-541:-LBgzSUS6cbru9fa7r8m:b-896fix) ([raw file](https://github.com/sillsdev/saymore/blob/09455e50f85a104e4f38e297478798865189adb4/src/SayMoreTests/model/SessionArchivingTests.cs#L541)):*
<details><summary><i>Previously, jasonleenaylor (Jason Naylor) wrote…</i></summary><blockquote>

_ in front of a variable indicates that it is private. So this should be metaFile or MetaFile
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/sillsdev/saymore/25)*

**Pull request merged**